clash-cores/src/Clash/Cores/Xilinx/BlockRam.hs

```diff
@@ -0,0 +1,97 @@
+module Clash.Cores.Xilinx.BlockRam (tdpbramBv, tdpbram) where
+
+import Clash.Explicit.Prelude hiding (enable)
+import Clash.Cores.Xilinx.BlockRam.Internal (tdpbram#)
+
+import GHC.Stack (HasCallStack)
+
+-- | Instantiates a Block Memory Generator IP as described in PG058.
+--
+-- Any value that is being written on a particular port is also the
+-- value that will be read on that port, i.e. the same-port read/write behavior
+-- is: WriteFirst. For mixed-port read/write, when port A writes to the address
+-- port B reads from, the output of port B is undefined, and vice versa.
+tdpbramBv ::
+  forall nAddrs domB domA nBytes .
+  ( HasCallStack
+  , KnownNat nAddrs
+  , KnownDomain domA
+  , KnownDomain domB
+  , KnownNat nBytes
+  ) =>
+
+  Clock domA ->
+  -- | Enable
+  Signal domA Bool ->
+  -- | Address
+  Signal domA (Index nAddrs) ->
+  -- | Write byte enable
+  Signal domA (BitVector nBytes) ->
+  -- | Write data
+  Signal domA (BitVector (8 * nBytes)) ->
+
+  Clock domB ->
+  -- | Enable
+  Signal domB Bool ->
+  -- | Address
+  Signal domB (Index nAddrs) ->
+  -- | Write byte enable
+  Signal domB (BitVector nBytes) ->
+  -- | Write data
+  Signal domB (BitVector (8 * nBytes)) ->
+
+  ( Signal domA (BitVector (8 * nBytes))
+  , Signal domB (BitVector (8 * nBytes))
+  )
+tdpbramBv = tdpbram
+{-# INLINE tdpbramBv #-}
+
+-- | Instantiates a Block Memory Generator IP as described in PG058.
+--
+-- Any value that is being written on a particular port is also the
+-- value that will be read on that port, i.e. the same-port read/write behavior
+-- is: WriteFirst. For mixed-port read/write, when port A writes to the address
+-- port B reads from, the output of port B is undefined, and vice versa.
+tdpbram ::
+  forall nAddrs domB domA nBytes a .
+  ( HasCallStack
+  , KnownNat nAddrs
+  , KnownDomain domA
+  , KnownDomain domB
+  , KnownNat nBytes
+  , BitSize a ~ (8 * nBytes)
+  , NFDataX a
+  , BitPack a
+  ) =>
+
+  Clock domA ->
+  -- | Enable
+  Signal domA Bool ->
+  -- | Address
+  Signal domA (Index nAddrs) ->
+  -- | Write byte enable
+  Signal domA (BitVector nBytes) ->
+  -- | Write data
+  Signal domA a ->
+
+  Clock domB ->
+  -- | Enable
+  Signal domB Bool ->
+  -- | Address
+  Signal domB (Index nAddrs) ->
+  -- | Write byte enable
+  Signal domB (BitVector nBytes) ->
+  -- | Write data
+  Signal domB a ->
+
+  ( Signal domA a
+  , Signal domB a
+  )
+tdpbram clkA enA addrA byteEnaA datA clkB enB addrB byteEnaB datB =
+  -- [Note: eta port names for tdpbram]
+  --
+  -- By naming all the arguments and setting the -fno-do-lambda-eta-expansion GHC
```

Comment:
Cool if this works.
But shouldn't you be setting this option then?

Reply:
That's a good point :see_no_evil:, thank you.